server/src/internalClusterTest/java/org/elasticsearch/search/basic/SearchRedStateIndexIT.java

```diff
@@ -52,7 +54,7 @@ public void testAllowPartialsWithRedState() throws Exception {
         SearchResponse searchResponse = client().prepareSearch().setSize(0).setAllowPartialSearchResults(true)
                 .get();
         assertThat(RestStatus.OK, equalTo(searchResponse.status()));
-        assertThat("Expect no shards failed", searchResponse.getFailedShards(), equalTo(0));
+        assertThat("Expect some shards failed", searchResponse.getFailedShards(), allOf(greaterThan(0), lessThanOrEqualTo(numShards)));
```

Comment:
Could we also validate that a failure is returned here in the failures field (as well as it being the right kind of failure)?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/64337/commits/1427b5c1dde9e716c161f296b46a2e50aea3342a